client/js/lounge.js

```diff
@@ -1107,16 +1108,25 @@ $(function() {
 	});
 
 	chat.on("input", ".search", function() {
-		var value = $(this).val().toLowerCase();
-		var names = $(this).closest(".users").find(".names");
-		names.find(".user").each(function() {
-			var btn = $(this);
-			var name = btn.text().toLowerCase().replace(/[+%@~]/, "");
-			if (name.indexOf(value) === 0) {
-				btn.show();
-			} else {
-				btn.hide();
-			}
+		const value = $(this).val();
+		const names = $(this).closest(".users").find(".names");
+
+		names.find(".user").each((i, el) => {
+			$(el).text($(el).text().replace(/<\/?b>;/, "")).hide();
+		});
+
+		const fuzzyOptions = {
+			pre: "<b>",
+			post: "</b>",
+			extract: el => $(el).text()
+		};
+
+		fuzzy.filter(
+			value,
+			names.find(".user").toArray(),
+			fuzzyOptions
+		).forEach(el => {
+			$(el.original).html(el.string).show();
```

Comment:
That's a lot of HTML shuffling imo, what if we had a separate container for results and rendered it once? This would also render #890 unnecessary.